DataFormats/PatCandidates/interface/CovarianceParameterization.h

```diff
@@ -0,0 +1,53 @@
+#ifndef _DataFormats_PatCandidates_CovarianceParameterization_h_
+#define _DataFormats_PatCandidates_CovarianceParameterization_h_
+#include <TFile.h>
+#include <TH3D.h>
+#include <iostream>
+#include <map>
+#include <TKey.h>
+class CompressionElement {
+    public:
+      enum Method {float16=0,reduceMantissa=1,logPack=2,tanLogPack=3,zero=4,one=5};
+      enum Target {realValue=0,ratioToRef=1,differenceToRef=2};
+      CompressionElement():method(zero),target(realValue){}
+      //CompressionElement(Method m, Target t, int bitsUsed, std::initializer_list<float> p): method(m),target(t),bits(bitsUsed),params(p){}
+      CompressionElement(Method m, Target t, int bitsUsed, std::vector<float> p): method(m),target(t),bits(bitsUsed),params(p){}
+      Method method;
+      Target target;
+      int bits;
+      std::vector<float> params;
+      uint16_t pack(float value, float ref=0.) const ;
+      float unpack(uint16_t packed, float ref=0.) const;
+
+};
+  
+
+class CovarianceParameterization {
+    public:
+        static int index(int i, int j) {if(i>=j) return j+i*(i+1)/2; else return i+j*(j+1)/2 ; }
+        struct CompressionSchema {
+             CompressionSchema() : elements(15,CompressionElement()) {}
```

Comment:
well, I read it says that if it has non trivial constructor (that should be the case) it is default initialized, not zero, do I understand it right?